torchvision/ops/focal_loss.py

```diff
@@ -32,6 +32,14 @@ def sigmoid_focal_loss(
         Loss tensor with the reduction option applied.
     """
     # Original implementation from https://github.com/facebookresearch/fvcore/blob/master/fvcore/nn/focal_loss.py
+
+    # Check reduction option
+    modes = ["none", "mean", "sum"]
+    if reduction not in modes:
+        raise ValueError(
+            f"Invalid Value for arg 'reduction': '{reduction} \n Supported reduction modes: 'none', 'mean', 'sum'"
+        )
```

Comment:
What is the reason for using a different idiom here? Why not do the same if/elif/else approach as with the other losses?

Reply:
My initial suggestion was to check the valid reduction mode even before we compute the losses. Throw value error immediately if it's badly configured.

Reply:
That make sense but that idiom requires us to maintain 2 separate lists with accepted modes. It also deviates from what all the other kernels are doing. It might be worth adopting a common strategy that aligns closely with Core. Thoughts?

Reply:
While aligning with core is OK. But it's a waste of computation. Compute the entire loss just to throw a valueError? 

And we could possibly refactor this a utility.

Reply:
> What is the reason for using a different idiom here? Why not do the same if/elif/else approach as with the other losses?

Before applying the reduction method to the final loss, we compute `ce_loss`  first (which requires a valid reduction value).
Were it not for using that specific code, we would have got repetitive blocks of if-else statements.

But, as far as the idiom is concerned we can stick with the if-else approach.

Reply:
> While aligning with core is OK. But it's a waste of computation. Compute the entire loss just to throw a valueError?
> 
> And we could possibly refactor this a utility.

Agree

Reply:
I thought we can use just this idiom and it does the job. I'm not getting how we need 2 lists

Reply:
Thanks for the comments, a few clarifications on my end:

> I'm not getting how we need 2 lists

We keep track the available options twice because the options are recorded once in the list (which you constantly reinitialise and look serially) for every call and once on the series of if/elif/else statements.

> But it's a waste of computation.

Should we be optimizing for the case of returning the error sooner? This seems to me an unnecessary optimization which leads to an idiom that is not present anywhere else in the codebase (TorchVision or Core). This is why I proposed to remove it.

Reply:
Fair enough. I should point out that maintaining a consistent structure overall will help new contributors like myself catch up with the codebase.